src/vorta/views/main_window.py

```diff
@@ -135,13 +135,22 @@ def set_icons(self):
         self.profileDeleteButton.setIcon(get_colored_icon('minus'))
         self.miscButton.setIcon(get_colored_icon('settings_wheel'))
 
-    def set_progress(self, text=''):
-        self.progressText.setText(text)
-        self.progressText.repaint()
+    def set_progress(self, profile_id, text=''):
+        profile = BackupProfileModel.get_by_id(profile_id)
+        profile.last_status = text
+        profile.save()
+        if profile.name == self.current_profile.name:
+            self.progressText.setText(text)
+            self.progressText.repaint()
 
     def set_log(self, text=''):
-        self.logText.setText(text)
-        self.logText.repaint()
+        profile = extract_profile_name(text)
```

Comment:
Why don't you modify the log signal similar to the progress signal. We shouldn't inexplicitely expect a specific log message format.